qiskit/providers/aer/jobs/utils.py

```diff
@@ -120,14 +120,20 @@ def _split_qobj(qobj, max_size, qobj_id, seed):
     return qobjs
 
 
-def _check_custom_instruction(experiments):
-
-    # check if save instruction exist
-    for exp in experiments:
-        for inst in exp.instructions:
-            if "save" in inst.name:
-                return True
-    return False
+def _check_custom_instruction(experiments, optypes=None):
+    """Return True if circuits contain instructions that cant be split"""
+    # Check via optype list if available
+    if optypes is not None:
+        # Optypes store class names as strings
+        return any(
+            "Save" in name or "Snapshot" in name
+            for optype in optypes for name in optype
+        )
+    # Otherwise iterate over instructions
+    return any(
+        "save" in inst.name or "snapshot" in inst.name
+        for exp in experiments for inst in exp.instructions
+    )
```

Comment:
This seems very fragile.  It'd be better to test against the exact names and classes - for `optype` it'd even likely be more efficient, because the complexity would scale with the number of "custom instruction types" rather than the number of instruction types in the circuit.

Reply:
Yeah, I would also like to remove this completely in the future since you should be able to support these instructions now that the type and subtype of save instructions is in metadata.